src/service.rs

```diff
@@ -39,25 +32,28 @@ impl UserService {
         Self { repository, config }
     }
 
-    fn create_token(&self, user_id: i64, email: &str) -> ServiceResult<String> {
-        let from_now = Duration::from_secs(3600);
-        let expired_future_time = SystemTime::now().add(from_now);
-        let exp = OffsetDateTime::from(expired_future_time);
-
-        let claims = Claims {
-            sub: String::from(email),
-            exp: exp.unix_timestamp() as usize,
-            user_id,
-        };
-
-        let token = encode(
-            &Header::default(),
-            &claims,
-            &EncodingKey::from_secret(self.config.token_secret.as_bytes()),
+    fn hash_password(&self, raw_password: &str) -> ServiceResult<String> {
+        let password_bytes = raw_password.as_bytes();
+        let hashed_password = argon2::hash_encoded(
+            password_bytes,
+            self.config.argon_salt.as_bytes(),
+            &Config::default(),
         )
-        .map_err(|err| ServiceError::InternalServerErrorWithContext(err.to_string()))?;
+        .unwrap();
```

Comment:
replace usage of unwrap with map_err as we would definitely want to avoid a panic here

Reply:
resolved